### PR TITLE
Add typing support for <Pressable /> RN 63-rc.0

### PR DESCRIPTION
**Pull request by ph1p**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/blob/44ec762e41029bf43530b1ff9b36ca3512c526e2/Libraries/Components/Pressable/Pressable.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by TheSavior** [comment]
In Pressable, the style and children props also take functions that get passed some state, like `pressed`.

**Comment by sandersn** [comment]
This change looks fine to me, but I don't like merging changes to such a widely used package without some authors signing off.

@alloy @huhuanming @iRoachie @timwangdev @kamal @alexdunne @swissmanu @bm-software @a-tarasyuk @mvdam @Esemesek @mrnickel @souvik-ghosh @nossbigg @saranshkataria @franzmoro @tykus160 @jakebloom @ceyhun @mcmar @theohdv @TheSavior @romain-faust @bebebebebe @Naturalclar @chinesedfan @vtolochk @SychevSP @RageBill @sasurau4 @256hz @doumart @drmas @akrger @jeremybarbet @ca057 @ds300 @natsathorn 
Can you take a look?

**Review by alloy — CHANGES_REQUESTED** [review]
@sandersn This should not be merged yet, as these are API additions for a RC, not a stable version. In reality this PR should have been a ‘draft’, but it doesn’t seem like you can convert it to one after the fact (or at least I can’t do so). I’ll mark it as needing changes, for now.

**Comment by alloy** [comment]
@ph1p Thanks 👍

**Comment by ph1p** [comment]
Changed it back to "draft".

This is actually a relatively new feature https://github.blog/changelog/2020-04-08-convert-pull-request-to-draft/

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -466,6 +468,79 @@ export interface Insets {
...
+    /**
+     * Whether the press behavior is disabled.
+     */
+    disabled?: boolean;
+
+    /**
+     * Additional distance outside of this view in which a touch is considered a
+     * press before `onPressOut` is triggered.
+     */
+    pressRectOffset?: number;
```
Comment (TheSavior) [inline types/react-native/index.d.ts:516]: Heads up that I’m planning on changing this in core today and picking it into 63. Changing it to pressRetentionOffset to be consistent with the touchables and Text

**Comment by pedpess** [comment]
This PR could be closed in favor of the one that was already merged in the mentions to avoid confusion

**Comment by alloy** [comment]
@pedpess Yup, good call 👍

**Pull request closed without merging**